sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -241,29 +249,35 @@ object functions {
    * @since 1.3.0
    */
   @deprecated("Use approx_count_distinct", "2.1.0")
-  def approxCountDistinct(e: Column): Column = approx_count_distinct(e)
+  def approxCountDistinct(e: Column): Column = withAggregateFunction {
```

Comment:
since we need to touch a lot of functions anyway, here is a long-standing cleanup work that is not finished: most of the functions can be implemented by creating `UnresolvedFunction`. We don't need to create the function expression directly.

Reply:
Ok, I will update the PR.

Reply:
https://github.com/apache/spark/pull/42864 is merged now and this PR is ready for review.